src/base/static/components/organisms/map-measurement-overlay.tsx

```diff
@@ -0,0 +1,438 @@
+/** @jsx jsx */
+import * as React from "react";
+import { css, jsx } from "@emotion/core";
+import PropTypes from "prop-types";
+import { featureCollection, point, lineString, polygon } from "@turf/helpers";
+import area from "@turf/area";
+import distance from "@turf/distance";
+import { geoPath, geoTransform } from "d3-geo";
+import WebMercatorViewport, {
+  WebMercatorViewportOptions,
+  WebMercatorViewport as IWebMercatorViewport,
+} from "viewport-mercator-project";
+import { CanvasOverlay } from "react-map-gl";
+import {
+  FeatureCollection,
+  Feature,
+  Point,
+  LineString,
+  Polygon,
+} from "geojson";
+
+import { mapViewportPropType } from "../../state/ducks/map";
+import { RegularText } from "../atoms/typography";
+import {
+  CreatePolygonIcon,
+  CreatePolylineIcon,
+  DeleteGeometryIcon,
+  UndoIcon,
+} from "../atoms/icons";
+
+const mapMeasurementOverlayProps = {
+  viewport: mapViewportPropType.isRequired,
+  width: PropTypes.number.isRequired,
+  height: PropTypes.number.isRequired,
+};
+
+interface ExposedCanvasOverlay extends CanvasOverlay {
```

Comment:
This might be a non-standard pattern-- I was trying to communicate that we're reaching into the `CanvasOverlay` type and exposing what's intended to be a private property.

Reply:
Yes, this is the standard pattern for extending interfaces. Accessing private properties from third-party libraries, however, is a red flag. But if we really need to access `_canvas`, then this seems like the proper way to type it :+1: